Workspaces/ProjectWorkspace/Scripts/AssetInstantiation.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using UnityEngine;
+using UnityEditor.Experimental.EditorVR.Data;
+
+namespace UnityEditor.Experimental.EditorVR.Utilities
+{
+    internal static class AssetInstantiation
+    {
+        const string k_AudioClipAttachUndoLabel = "Add Audio Clip";
+        const string k_AssignFontUndoLabel = "Assign Font";
+        const string k_AttachScriptUndoLabel = "Add Script";
+        const string k_AssignMaterialUndoLabel = "Assign Material";
+        const string k_AssignPhysicMaterialUndoLabel = "Assign Physic Material";
+        const string k_AssignMaterialShaderUndoLabel = "Assign Material Shader";
+
+        internal static AudioSource AttachAudioClip(GameObject go, AssetData data)
+        {
+            Undo.RecordObject(go, k_AudioClipAttachUndoLabel);
```

Comment:
Are you sure this is the proper way to do the undo here? I figured we would do a RegisterCreatedObject on the new component, otherwise, just setting the field and incrementing the group should do? Might we still need to RegisterCreatedObject separately?

Reply:
Also, thanks for thinking about undo support 👍

Reply:
everything must Undo :)